### ISSUE TITLE
TypeError: prevDeps is undefinded, when editing content

### ISSUE THREAD

**Issue by cleitonper** [tinacms/tinacms#611]
## Description

After adding the TinaCMS in my existing Gatsby site, I'm receiving a error message when editing my content both in TinaCMS `sidebar` and directly in my `markdown` files. When I remove TinaCMS dependencies the error does not occurs anymore.

The error says the following:

```
TypeError: prevDeps is undefinded 
```

### Steps to reproduce

* install tina-cms to my existing gatsby project
* setup it in `gatsby-config.js`
* run a development server with the `gatsby develop` command
* access `http://localhot:8000/` in my browser.
* edit a content in the tina sidebar

### Expected result

Edit the content without errors.

### Actual result
After editing the content a error message is shown.

#### Tinacms Package Versions

```text
 gatsby-plugin-tinacms@0.1.19
│ ├── @tinacms/core@0.6.0
│ └─┬ tinacms@0.11.2
│   ├─┬ @tinacms/fields@0.2.1
│   ├─┬ @tinacms/forms@0.1.0
│   ├── @tinacms/icons@0.3.9
│   ├─┬ @tinacms/react-core@0.2.3
│   ├── @tinacms/styles@0.1.0
├─┬ gatsby-tinacms-git@0.2.16
│ ├─┬ @tinacms/api-git@0.5.2
│ ├── @tinacms/git-client@0.4.1
├─┬ gatsby-tinacms-remark@0.6.3│ ├─┬ @tinacms/form-builder@0.2.11
```

#### Tools

| Tool    | Version |
| -------- | ---------- |
| browser | Firefox Developer Edition v72.0b3  |
| os          |  Manjaro 18.1.4 Juhraya |
| yarn       | v1.17.3 |
| node      |  v10.16.0  |
| npm       | v6.10.2  |

#### URL to the project repository

[`https://github.com/cleitonper/me`](https://github.com/cleitonper/me/tree/chore/tina-cms)

The TinaCMS is used in the home page ([`/src/pages/index.tsx`](https://github.com/cleitonper/me/blob/chore/tina-cms/src/pages/index.tsx))

The content that I'm trying to edit is inside this file: [`src/content/presentation/presentation.md`](https://github.com/cleitonper/me/blob/chore/tina-cms/src/content/presentation/presentation.md)

**Note:** use the `tina-cms` branch

#### Error Screenshot

| Error Message |
| :------------------: |
| ![Error](https://user-images.githubusercontent.com/13934790/72506186-2f2feb80-3820-11ea-8a17-48ed1f92a0e1.png) |
| `TypeError: prevDeps is undefined` |

**Comment by ncphillips** [comment]
Thanks for the report @cleitonper. The link to your repo will be helpful. I'll dig into troubleshooting this shortly.

**Comment by ncphillips** [comment]
Debugging Notes:
* Forked the site and was quickly able to reproduce this issue. 
* Confirmed that this is happening with latest versions of all tinacms packages
* Noticed that this site is using styled-components v5. Probably not an issue.
* Using latest version react: `16.12.0`. Tried downgrading to 16.9 but same issue.
* Added a log at the top of the component:
   ```ts
   const HomePage: FunctionComponent<Props> = ({ data }) => {
     console.log("Presentation", data.presentation);
     const [presentation] = useLocalRemarkForm(data.presentation);
  ```
  When the error is thrown the log shows that `data.presentation` is null: <img width="498" alt="image" src="https://user-images.githubusercontent.com/824015/72539199-5359ef80-3855-11ea-907f-1b188f54ba45.png">
  This is definitely the event that is causing the error. Honestly, an error or at least some janky behavior seems like a pretty understandable response to suddenly not being given any data. The ideal fix would be for Gatsby to not do this. Still, I'm going to take a look at the Tina hooks to see this can be handled more elegantly.

**Comment by ncphillips** [comment]
* Cloned project in to the `packages/@testing` directory. 
  In order to make that work I had to delete `gatsby` from `me/node_modules`
  After doing this the bug disappeared. 
* Switched back to the `me` project and pinned `gatsby@2.15.34`
  Bug disappeared. **This issue is likely caused by a regression in newer versions of Gatsby**
  A bug report should be opened with gatsby, but we should also explore a fix from our end.

**Comment by DirtyF** [comment]
Stumbled on a similar issue with Tina Grande starter on Gatbsy Cloud, when ever I start typing in a field, Gatsby just explodes 💥 

<img width="1552" alt="prevDeps-error" src="https://user-images.githubusercontent.com/103008/72546557-3dffb800-388b-11ea-8be7-53ad5e3f4f5d.png">

**Comment by nibtime** [comment]
I upgraded Gatsby to `2.18.21` and experienced the same issue + other issues (like characters are skipped or rearranged while editing). The `data` object of queries was `null` sometimes and yielded the the `remarkForm` not configured correctly error message because of that.

After I downgraded again everything worked fine, so there seems to be something going on in newer versions. Would be interesting to know what versions introduced those regressions, because this can potentially kill the great first impressions Tina will offer if everything works.

**Comment by ncphillips** [comment]
@nibtime I've made a prerelease that might fix one of the issues. Could you try that one out with the versions of gatsby that are broken for you?

```
yarn add gatsby-plugin-tinacms@next gatsby-tinacms-git@next gatsby-tinacms-remark@next gatsby-tinacms-json@nex
```

**Comment by ncphillips** [comment]
@all-contributors please add @cleitonper for bugs

**Comment by cleitonper** [comment]
Thanks for the quick response @ncphillips.

After downgrading `gatsby` to `v2.15.34` the *bug* disappear when updating the content, but i noticed that when I let the `title` field empty it's come back again.

opening the `presentation.md` file, where the content is stored, I can see that `tina` or `gatsby` broke the structure of the file.

before editing I have this:

```
---
title: Apresentação
---
```

after I let the field empty in *tina sidebar* the file is changed to this:

```
---
!<tag:yaml.org,2002:js/undefined> ''
---
```

___

I'll try to install the prerealeses versions mentioned here.

**Comment by ncphillips** [comment]
Hey @cleitonper, I just made a release. Could you try again with `latest`? I think it may have fixed the above issue as well. The before, the title would have been removed from the frontmatter when you cleared the field, but that has been addressed in c899ee5

**Comment by cleitonper** [comment]
This issue is fixed at latest version.

Thanks @ncphillips and also the whole TinaCMS team.

___

<details>
<summary>used versions</summary>
<p>

```json
"gatsby-plugin-tinacms": "^0.1.20",
"gatsby-tinacms-remark": "^0.6.4",
"gatsby-tinacms-git": "^0.2.17",
```

</p>
</details>
